### ISSUE TITLE
Content: OMOP mappings `-->` SSSOM `-->` FHIR ConceptMap uploads

### ISSUE THREAD

**Issue by joeflack4** [HOT-Ecosystem/hapi-issues#59]
## Overview
This is something Stephanie is working on and will result in a lot of content that we can upload to our server (I think 100+ SSSOMified ConceptMaps). I think that this is something that Chris Chute would like (and may have even alluded to) to have uploaded to our server.

## Related
- [TIMS/N3C: [OMOP <--> ICD10CM, etc] --> SSSOM --> FHIR ConceptMap #140](https://github.com/jhu-bids/TermHub/issues/140): I should have made the issue under TIMS, but I made it under TermHub for no good reaosn. The issue is described in full there (w/ discussion). Stephanie and I are actively working on and making fast progress.
- https://github.com/HOT-Ecosystem/hapi-issues/issues/104

**Comment by joeflack4** [comment]
Not really related to the connectathon in any particular way other than having more content, but added it to that board since it's actively in progress.

**Comment by chrisroederucdenver** [comment]
Chris Chute's concern about the connectathon is to not get overly focussed on the Genomics or Phenopackets IGs, so I can see where this might be of interest. Can you summarize the content that will end up in the FHIR server?

Is this a mapping from ICD10 codes to OMOP concept_ids mined out of SQL, put into SSSOM so we can translate it to FHIR? 
 I'm still not sure how this data would be used, other than pulling it out of the server and going back where it came from. Is there a more direct use-case for OMOP mappings in FHIR? This does sound like a requirement for terminology services in general, just not the connectathon. So your initial assessment of putting the ticket in term-hub doesn't sound too bad.
 
Addressing Chris's point at the connectathon would require having a look at other tracks and their IG's to see what data and what terminologies, value sets and mappings they are using. With that data we can get content and engage the folks running those tracks.

**Comment by joeflack4** [comment]
@chrisroederucdenver Stephanie is doing most of the work on this. I'm helping her out with SSSOM wrangling, and will ultimately run the conversion and upload to our server. Most of my effort on this is coming out of my 7.5% LinkML/TCCM/SSSOM FTEs and not my 17.5% TIMS FTEs.

From what I understand, this is using OMOP as a middleman to get mappings between other ontologies, e.g. ICD10CM and SNOMED.

Unfortunately I don't know the motivation for getting this mapping content through OMOP specifically. I just know that Chris wants it. That's kind of how it is for 90% of these issues. I think it still hasn't been generally specified how the server and its content will be ultimately used... yet!

**Comment by stephanieshong** [comment]
@joeflack4 - can you assign me to this? I am not able to self assign

**Comment by chrisroederucdenver** [comment]
done

From: Stephanie Hong ***@***.***>
Date: Tuesday, October 4, 2022 at 11:26 AM
To: HOT-Ecosystem/hapi-fhir-jpaserver-starter ***@***.***>
Cc: Roeder, Chris3 ***@***.***>, Mention ***@***.***>
Subject: Re: [HOT-Ecosystem/hapi-fhir-jpaserver-starter] OMOP mappings --> SSSOM --> FHIR ConceptMap uploads (Issue #59)
[External Email - Use Caution]

@joeflack4<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjoeflack4&data=05%7C01%7Cchris.roeder%40cuanschutz.edu%7Cb6d0f69e5a8a4c07567308daa62d9420%7C563337caa517421aaae01aa5b414fd7f%7C0%7C0%7C638005011933188632%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=a76nmSfWVQUF5Jy9biZYsrbQQLVEUO8bflyVyLWrY1o%3D&reserved=0> - can you assign me to this? I am not able to self assign

**Comment by stephanieshong** [comment]
@chrisroederucdenver - thank you.

**Comment by joeflack4** [comment]
I think this one is currently done. We can always reopen if we need to do another iteration of this.

First implementation was direct OMOP to FHIR:
- https://github.com/jhu-bids/omop-vocab-on-fhir

Has now been superseded by approach using OWL as an intermediary:
- https://github.com/HOT-Ecosystem/omop2fhir-vocab

However, it is possible that this process of using an OWL intermediary might also be lossy. No comparison has been made yet between these two methods, so it is possible we may reopen the original implementation at some point, or probably better to upgrade the OWL implementation.

**Comment by stephanieshong** [comment]
@ShahimEssaid @joeflack4  -If useful we can resume and provide you with additional concept map resource content you can add to the HAPI server, pick up from where we left off.
